build/bp-messages/bp-messages-blocks.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * BP Friends Blocks Functions.
+ *
+ * @package   bp-blocks
+ * @subpackage \build\bp-friends\bp-friends-blocks
+ */
+
+namespace BP\Blocks;
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Register the friends blocks.
+ *
+ * @since 9.0.0
+ */
+function register_messages_blocks() {
+	$blocks = array(
+		'bp/friends' => array(
+			'name'               => 'bp/sitewide-notices',
+			'editor_script'      => 'bp-sitewide-notices-block',
+			'editor_script_url'  => plugins_url( 'js/blocks/sitewide-notices.js', __FILE__ ),
+			'editor_script_deps' => array(
+				'wp-blocks',
+				'wp-element',
+				'wp-components',
+				'wp-i18n',
+				'wp-editor',
+				'wp-block-editor',
+				'bp-block-data',
+			),
+			'style'              => 'bp-sitewide-notices-block',
+			'style_url'          => plugins_url( 'css/blocks/sitewide-notices.css', __FILE__ ),
+			'attributes'         => array(
+				'title' => array(
+					'type'    => 'string',
+					'default' => '',
+				),
+			),
+			'render_callback'    => __NAMESPACE__ . '\bp_messages_render_sitewide_notices_block',
+		),
+	);
+
+	foreach ( $blocks as $block ) {
+		bp_register_block( $block );
+	}
+}
+add_action( 'bp_messages_blocks_init', __NAMESPACE__ . '\register_messages_blocks', 10, 0 );
+
+/**
+ * Callback function to render the BP Friends Block.
+ *
+ * @since 9.0.0
+ *
+ * @param array $attributes The block attributes.
+ * @return string           HTML output.
+ */
+function bp_messages_render_sitewide_notices_block( $attributes = array() ) {
+	$block_args = bp_parse_args(
+		$attributes,
+		array(
+			'title' => '',
+		),
+		'widget_object_sitewide_messages'
+	);
+
+	if ( ! is_user_logged_in() ) {
+		return;
+	}
+
+	// Don't display the block if there are no Notices to show.
+	$notice = \BP_Messages_Notice::get_active();
+	if ( empty( $notice ) ) {
+		return;
+	}
+
+	$closed_notices = bp_get_user_meta( bp_loggedin_user_id(), 'closed_notices', true );
+
+	if ( empty( $closed_notices ) ) {
+		$closed_notices = array();
+	}
+
+	if ( is_array( $closed_notices ) && in_array( $notice->id, $closed_notices )  ) {
```

Comment:
Maybe do `$closed_notices = (array) bp_get_user_meta( bp_loggedin_user_id(), 'closed_notices', true );`?

Removing the conditional.

Reply:
Yeah +1

Reply:
Sure! Maybe we should change the code in BP where I got that logic from, too.
`bp_message_get_notices()` and `bp_nouveau_push_sitewide_notices()` are similar.